File: Easy/20-Valid Parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let charStack = [];
    charStack.push(s[0])
    for(let i = 1; i<s.length; i++){
        let top = charStack[charStack.length - 1];
        if((top === '(' && s[i] === ')') || (top === '{' && s[i] === '}') || (top === '[' && s[i] === ']')){
            charStack.pop()
        }else{
            charStack.push(s[i])
        }
    }
    if(charStack.length === 0){
        return true
    }else{
        return false
    }
};

File: Easy/20-Valid Parentheses.c
bool isValid(char* s) {
    char arr[10000];
    // Better for space complexity
    // int n = strlen(s);
    // char* arr = (char*)malloc(n);
    int top=0, i=0;
    while(s[i]!='\0')
    {
        if(top==0)
        {
            arr[top] = s[i];
            top++;
        }else
        {
            if(s[i] == ')' && arr[top-1] == '(')
            {
                top--;
            }else if(s[i] == ']' && arr[top-1] == '[')
            {
                top--;
            }else if(s[i] == '}' && arr[top-1] == '{')
            {
                top--;
            }else{
                arr[top] = s[i];
                top++;
            }
        }
        i++;
    }
    if(top == 0)
    {
        return true;
    }else
    {
        return false;
    }
}
